ccmlib/utils/download.py

```diff
@@ -148,6 +148,10 @@ def download_version_from_s3(url: str, target_path: str, verbose=False):
             logging.warning(f"url: '{url}' wasn't found on S3")
             logging.warning(f"download might be very slow")
             return None
+        elif 'Forbidden' in str(ex):
```

Comment:
instead of inspecting the stringified representation of the exception, i'd suggest follow [AWS's document](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/error-handling.html#parsing-error-responses-and-catching-exceptions-from-aws-services), and compare the error code with the ones defined by the [document](https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html#ErrorCodeList) like:
```py
except botocore.client.ClientError as ex:
  error_code = error.response['Error']['Code']
  if error_code in ('NoSuchBucket', 'NoSuchKey'):
    logging.warning(f"url: '{url}' wasn't found on S3")
    logging.warning(f"download might be very slow")
    return None
  elif error_code == "AccessDenied"
    logging.warning(f"url: '{url}' Forbidden (403) on S3")
    logging.warning(f"download might be very slow")
    return None
  else:
    raise
```

Reply:
replaced it